### ISSUE TITLE
Authorized Failed

### ISSUE THREAD

**Issue by graccelle** [Azure/azure-cli#29354]
### Describe the bug

Trying to run my python script, trying to access the azure OpenAI service using my API credentials, but keep getting the "authorization failed" error message.

### Related command

azure openai

### Errors

azure.core.exceptions.HttpResponseError: () Authorization failed.
Code:
Message: Authorization failed.

### Issue script & Debug output

![Screenshot 2024-07-08 161944](https://github.com/Azure/azure-cli/assets/54447737/fb486776-974a-4b2d-9c03-0969028ed536)

### Expected behavior

Given the successful access to the chatbot local host (http://localhost:7860/), once i send a prompt, I should get a response to my question (like you would in ChatGPT). See attached screenshot of the error message i get instead on the user interface.
![Screenshot 2024-07-08 162737](https://github.com/Azure/azure-cli/assets/54447737/ba6e2348-fa8a-44c0-9852-a99b6710e37d)

### Environment Summary

Name: azure-core
Version: 1.30.2
Summary: Microsoft Azure Core Library for Python
Home-page: https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/core/azure-core
Author: Microsoft Corporation
Author-email: [azpysdkhelp@microsoft.com](mailto:azpysdkhelp@microsoft.com)
License: MIT License
Location: c:\users\t-grbabalola\appdata\local\anaconda3\envs\chatosp\lib\site-packages
Requires: requests, six, typing-extensions
Required-by: azure-identity, azure-search-documents, msrest

### Additional context

I have raised this issue before, and it was closed without being resolved. Hoping for a better outcome this time around and not just redirecting without any concrete solution.

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by graccelle** [comment]
I would appreciate a teams call if possible, incase this ticket is not detailing the problem I am facing.

**Comment by graccelle** [comment]
> Hi @graccelle Find similar issue #29334.
> 
> Issue title	Authorization Failed
> Create time	2024-07-08
> Comment number	0
> Please confirm if this resolves your issue.

No, it doesn't resolve my issue as per my initial comment.

**Comment by graccelle** [comment]
> Is this issue related to Azure CLI commands?

Yes it is.

**Comment by yonzhan** [comment]
Please provide detailed CLI commands.

**Comment by graccelle** [comment]
> Please provide detailed CLI commands.

Kindly find below:

(chatosp) C:\Users\t-grbabalola\PhishResearch\sources\DataScience\ChatOSP>az login
Error loading command module 'mysql': cannot import name 'StorageRedundancyEnum' from 'azure.mgmt.rdbms.mysql_flexibleservers.models._my_sql_management_client_enums' (C:\Program Files\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\mgmt\rdbms\mysql_flexibleservers\models\_my_sql_management_client_enums.pyc)
Error loading command module 'rdbms': cannot import name 'StorageRedundancyEnum' from 'azure.mgmt.rdbms.mysql_flexibleservers.models._my_sql_management_client_enums' (C:\Program Files\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\mgmt\rdbms\mysql_flexibleservers\models\_my_sql_management_client_enums.pyc)
Select the account you want to log in with. For more information on login with Azure CLI, see https://go.microsoft.com/fwlink/?linkid=2271136

Retrieving tenants and subscriptions for the selection...

[Tenant and subscription selection]

No      Subscription name                     Subscription ID                       Tenant
------  ------------------------------------  ------------------------------------  ---------
[1]     ADF Test sub - App Model V2           0ee78edb-a0ad-456c-a0a2-901bf542c102  Microsoft
[2]     AEP_CorePlatform_Playground_Dev       84ca48fe-c942-42e5-b492-d56681d058fa  Microsoft
[3]     AEther Dev                            e75c95f3-27b4-410f-a40e-2b9153a807dd  Microsoft
[4]     AISC-DEV-02                           b2a328a7-ffff-4c09-b643-a4758cf170bc  Microsoft
[5]     AISC-EngSys-01                        eef8b6d5-94da-4b36-9327-a662f2674efb  Microsoft
[6]     AKS Global corpdev                    f7616b56-4c4f-481f-8a80-0e0f51d430c5  Microsoft
[7]     AKS Global corptest                   9b89d16e-d816-4df5-9ba4-a527dc23af3c  Microsoft
[8]     AmbaUSX_MicroServices_Staging_US_...  d7cb4225-a5fb-45a5-9a19-74142ea65731  Microsoft
[9]     AmbaUSX_MicroServices_Staging_US_...  1993e859-2915-4812-b16a-04a3e755c27e  Microsoft
[10]    AML - Responsible AI R&D              e0fd569c-e34a-4249-8c24-e8d723c7f054  Microsoft
[11]    AML Infra - Engineering System        f1d79e73-f8e3-4b10-bfdb-4207ca0723ed  Microsoft
[12]    AnE.ExP.Production                    dad45786-32e5-4ef3-b90e-8e0838fbadb6  Microsoft
[13]    Applied Sciences Group Dev            e686ef8c-d35d-4e9b-92f8-caaaa7948c0a  Microsoft
[14]    ARM Test Environment                  2145a411-d149-4010-84d4-40fe8a55db44  Microsoft
[15]    ASI Hunting Demo Environment          40dcc8bf-0478-4f3b-b275-ed0a94f2c013  Microsoft
[16]    ASZ_Lab_Hardware_Dev_1                e2a8a7b3-cb7a-4930-9c79-2b0eb7bf843c  Microsoft
[17]    AzPIESecurity-DEV-Apple               4c33a6bb-d566-4d0b-8951-00b55a91fba8  Microsoft
[18]    Azure Carbon Testing                  326daf21-cbc6-40c8-877f-10324177771e  Microsoft
[19]    Azure CloudES CDP ER - Test           b65b516b-415b-4c68-a254-bfa7411275f8  Microsoft
[20]    Azure Portal Telemetry Reporting      4bbecc02-f2c3-402a-8e01-1dfb1ffef499  Microsoft
[21]    Azure Profiler - Testing              432a7068-99ae-4975-ad38-d96b71172cdf  Microsoft
[22] *  Azure Security Data Science Devel...  0bd0a0b9-12e9-4949-9011-38eb480fdf64  Microsoft
[23]    Azure System Lockdown Internal No...  d2a3f240-56f3-45bf-a0be-1d497a230cf4  Microsoft
[24]    Babel                                 a1920ebd-59b7-4f19-af9f-5e80599e88e4  Microsoft
[25]    BAP IA EU Schrems (P)                 b43a6159-1bea-4fa2-9407-e875fdc0ff55  Microsoft
[26]    Bing MM Measurement                   ae71ef11-a03f-4b4f-a0e6-ef144727c711  Microsoft
[27]    CaseManagement.Staging                7db5561b-ba02-450a-8af0-795c4755dc9e  Microsoft
[28]    CloudAnalytics_Prod1                  8ef80208-601f-4c83-802e-751d211745f9  Microsoft
[29]    Cog Services Computer Vision          f7b4a5ce-d3bd-4a03-84fe-8bcad772c8c1  Microsoft
[30]    Core-ES-BLD-Allegro-Dev               2998c1d7-09ae-4b08-b6d0-9c2ee5a84c35  Microsoft
[31]    Core-ES-BLD-Allegro-Prod              551f97b7-e9a4-4b2a-9be9-1f3c4db41b65  Microsoft
[32]    Core-ES-Signing-Dev                   3f01c831-e426-4e17-99dd-de39d98c4ead  Microsoft
[33]    Core-ES-STREAM-DeXDI                  38dbd059-bd26-4172-af17-bcc5a00744de  Microsoft
[34]    Core-ES-WorkManagement                52a442a2-31e9-42f9-8e3e-4b27dbf82673  Microsoft
[35]    COSMIC AKS PlayGround                 39675fbf-5b47-472e-9bb9-5570c6edbd4f  Microsoft
[36]    Cosmos_C&E_Azure_AzureEngineering...  8c4b5b03-3b24-4ed0-91f5-a703cd91b412  Microsoft
[37]    Cosmos_C&E_C&E Security Eng_Netwo...  adeb0103-9ba5-48dd-ad60-d087507cd63d  Microsoft
[38]    CSX-WSD-CFE-TNT-QPP                   44074499-ad9f-45b9-9b0b-f594f8736a45  Microsoft
[39]    CSX-WSD-CFE-TNT-TOOLKITHUB            3cd01953-be46-4f9b-a4e0-26f19f428765  Microsoft
[40]    Dakota-Dev                            188c28e4-46df-4487-ba50-89b5b644ab39  Microsoft
[41]    Data Pillar Security Tooling          60214a3c-65a2-46f8-8b32-f959f2454106  Microsoft
[42]    Defender_Infra_Test_AzSC              a5f808c3-0a4d-49c5-98d7-7259a02e5276  Microsoft
[43]    Developer VM Optimization             e188d541-aa57-45e8-8687-c83dc57a671e  Microsoft
[44]    Efficiency Pack Services (NonProd)    f7c445af-a4de-4264-9e87-3196d6bc384d  Microsoft
[45]    emilyzhu dev subscription             86f010b8-6473-4cb9-90c1-671c74faa4ee  Microsoft
[46]    ES-TAP-TestContent                    52942f45-54fd-4fd9-b730-03d518fedf35  Microsoft
[47]    ES_DeveloperProductivity_Dev          4628298e-882d-4f12-abf4-a9f9654960bb  Microsoft
[48]    HPCScrub1                             d2c9544f-4329-4642-b73d-020e7fef844f  Microsoft
[49]    IDEAs MS Reporting                    bac420ed-c6fc-4a05-8ac1-8c0c52da1d6e  Microsoft
[50]    IDXXpertRandD Dev Data Catalog        50ff7bc0-cd15-49d5-abb2-e975184c2f65  Microsoft
[51]    Internal App Insights Resources f...  3cd95ff9-ac62-4b5c-8240-0cd046687ea0  Microsoft
[52]    ITSM Integration - Production - New   6a0ab98c-24d4-4131-8cd7-0a78bfce58f5  Microsoft
[53]    LA_VMInsights_Telemetry_All_NonProd   526094ba-add2-4793-905d-a99153306c9d  Microsoft
[54]    LinuxMdsd Test                        13723929-6644-4060-a50a-cc38ebc5e8b1  Microsoft
[55]    Marvel Intelligence Model Trainin...  98b02a69-28a5-4ee7-a622-cd69c7a59c4e  Microsoft
[56]    MDE-SigRel-Services-PPE-1             8a1dc480-1e88-44b4-8f1e-4704973beed6  Microsoft
[57]    MedicalImagingAI Dev                  6c180dd2-1ec4-4fad-8ba8-1f2d8d67c129  Microsoft
[58]    Microsoft Azure Internal Consumpt...  1278a874-89fc-418c-b6b9-ac763b000415  Microsoft
[59]    MPS_MDA_Infra_MSFT_103                b4847c1c-6842-4621-ac95-e3bc5d2eb3fc  Microsoft
[60]    MPS_MDA_Infra_MSFT_STG                855e349b-2ff3-4031-87fd-df653a0fc249  Microsoft
[61]    MPS_MDA_Infra_MSFT_SWA_101            02b87eab-08e3-4443-af98-36e9b18aa65d  Microsoft
[62]    MPS_MDI_Microservice_Staging_US_C...  59f7872a-fcc9-4651-b7e1-f935bb9d55d3  Microsoft
[63]    MPS_MDI_Microservice_Staging_US_E...  dc271c9c-7dc4-4525-ae78-52bdbfe4883f  Microsoft
[64]    MSFT-Modern Device-Modern Mgmt-Im...  29de2cfc-f00a-43bb-bdc8-3108795bd282  Microsoft
[65]    MSFT-MVD-02-Staging                   0850ace2-594e-4326-bc8e-48b2d5bd5860  Microsoft
[66]    MSFT-WindowsVirtualDesktop-01         54b875cc-a81a-4914-8bfd-1a36bc7ddf4d  Microsoft
[67]    MTP_BAND_STG_EUS                      821eb853-2617-44be-935b-0f88b3b0b0cd  Microsoft
[68]    NetIso NRMS Microsoft Tenant Admi...  2c505610-a5dd-473e-aa33-b31aac398e29  Microsoft
[69]    NetworkIsolationMSFTSLNMManagement01  531c5c86-c959-41ec-91ff-5404a0b217b8  Microsoft
[70]    NetworkIsolationMSFTSLNMManagement02  d1cb9e0d-697e-40cf-bc22-53501ce396c3  Microsoft
[71]    nonprod.dex.jpn                       e2535072-195e-4bf0-a42e-132be61ffb42  Microsoft
[72]    nonprod.md.cpv2.fp.k8s.us.0           5d21c8a9-ec89-4c88-91b7-657f86b700e9  Microsoft
[73]    nonprod.mdatp.dataplatforms.0         943db234-ecc9-4326-925b-9edfec41f1bc  Microsoft
[74]    nonprod.wd.keyvault.research.0        68bdc264-eab0-48c5-80cb-ceb3b8185e47  Microsoft
[75]    O365 OFE - Focus Infrastructure (...  f1251103-af5a-44b7-ac2a-4e5538e80cde  Microsoft
[76]    Observability_AzMon_Grafana_Dev       4dc2cd39-7a89-43d8-bebe-8bb501359891  Microsoft
[77]    Observability_Grafana_Partners_Ho...  181685e0-3360-41ea-8ac3-9007a3a2db00  Microsoft
[78]    ODC-OneBox-Resources                  a471d615-ff98-4e80-b375-a19543d4691e  Microsoft
[79]    OE-AppleFundamentals - External -...  3bc94887-5f56-4d55-a6a0-de02cbfe8935  Microsoft
[80]    PJ-1SOC-01                            1af9f31d-d843-4aae-9954-5de652a9631a  Microsoft
[81]    PJ-MDO-01                             26173fda-914d-4a6e-ae9a-27f56418421f  Microsoft
[82]    Python_AI_Tools_PM_Team_Sub (jbin...  1163fbbe-27e7-4b0f-8466-195fe5417043  Microsoft
[83]    Pytorch Build                         3905431d-c062-4c17-8fd9-c51f89f334c4  Microsoft
[84]    Quantum Network Infrastructure        c7f8dda6-0f53-4afe-95ac-b53dcbbdb8ea  Microsoft
[85]    Responsible & OpenAi Research         3f577935-3138-4d07-86b3-75651b696483  Microsoft
[86]    Rome ILDC - IoT - Dev                 81de6a21-8799-4bb5-a28c-cbc292f8e023  Microsoft
[87]    Sentinel - Hunting - Test             f117ae3d-d6b2-4031-8d76-e0ca15e838ab  Microsoft
[88]    Sentinel - UEBA Playground            39ee335e-75c8-4529-be56-8bb96162e198  Microsoft
[89]    Sentinel Hunting Telemetry            89a62b27-571a-46aa-80b0-945d19cdfc09  Microsoft
[90]    Sentinel ILDC - Dev                   78ffdd91-611e-402f-8a7e-7ab0b209b7c6  Microsoft
[91]    Sentinel ILDC - E2E                   b12de5cd-8652-4157-85ba-079a79bb8074  Microsoft
[92]    Sentinel ILDC - Playground            de5fb112-5d5d-42d4-a9ea-5f3b1359c6a6  Microsoft
[93]    Sentinel ILDC - Scale Testing         90d46a29-399b-458b-b1ee-5cdb0f2d90e4  Microsoft
[94]    Sentinel ILDC - Telemetry             ef6ab2ac-7a70-4872-9694-7ee0bfad88da  Microsoft
[95]    Sentinel ILDC – Playground – Hori...  0563d7b3-7acf-4f1c-ade4-b332de0e9e0f  Microsoft
[96]    SentinelCAT Eco Dev                   4383ac89-7cd1-48c1-8061-b0b3c5ccfd97  Microsoft
[97]    SentinelUS Recommendations Dev        1456d992-8d50-4a7e-ae0d-59aa4028f61b  Microsoft
[98]    SentinelUS TI Matching Telemetry      7f0d9100-9be3-4981-b127-611338694353  Microsoft
[99]    SentinelUS-Playground                 17f076ee-acaf-43e4-8c93-55f60fa8d9e4  Microsoft
[100]   SentinelUS-TI-Augusta-Dev             45dc160c-4eea-41ef-b1bc-dd7dc2e77a72  Microsoft
[101]   SentinelUS-TI-TrafficAnalyzer         e8c90aa4-a2d6-43dc-b16a-544c8cd164f8  Microsoft
[102]   SentinelUS.Billing.Staging            927b5eaf-9948-43ca-80d7-f8894c661d63  Microsoft
[103]   SentinelUS.ContentManagement.Dogfood  54976ca6-57cf-4f73-afdd-f0ebee6cd7ae  Microsoft
[104]   SentinelUS.ContentManagement.GSM....  e1a832fd-b3f8-4bdb-81d2-f3302c81c363  Microsoft
[105]   ServicesPortfolio MCS                 eec2de82-6ab2-4a84-ae5f-57e9a10bf661  Microsoft
[106]   Skype-NetEM-STAGING                   8cff5d56-95fb-4a74-ab9d-079edb45313e  Microsoft
[107]   Sonar                                 a64a0d22-c949-42a1-ab7c-ddad6e604b55  Microsoft
[108]   Sonar Configuration Test              f33a3740-66ce-4d00-9923-4ef3595f6d25  Microsoft
[109]   SONAR.DAAS.DEV                        673e0411-d281-404d-be24-b377231eed79  Microsoft
[110]   SONAR.DAAS.RP                         8398b3fe-da23-4139-9c11-612faa036723  Microsoft
[111]   SONAR.DEV.TOOLSET                     c23b0c0c-1b9d-40c3-a9b8-a057fa21f5c8  Microsoft
[112]   spectre.8ks.dev                       17afca26-a0ef-4fab-bb35-69079b1b3caa  Microsoft
[113]   Speech Services - DEV - SDK (carbon)  e72e5254-f265-4e95-9bd2-9ee8e7329051  Microsoft
[114]   Speech Services - EXP2                a1c3dc6b-8630-4bb7-a29e-4ed4407c329b  Microsoft
[115]   SPO-CyrusB                            8ae1955e-f748-4273-a507-10159ba940f9  Microsoft
[116]   Terraform Testing on Azure with T...  85b3dbca-5974-4067-9669-67a141095a76  Microsoft
[117]   TEST - IaaS UX - Dev Tools            47013160-2f48-4226-8331-256b8c5cad14  Microsoft
[118]   TEST - IbizaFx misc                   8168a4f2-74d6-4663-9951-8e3a454937b7  Microsoft
[119]   Tip Node Service                      73c84423-276c-4f33-8048-2225a4ade5e7  Microsoft
[120]   TScience                              9ec1d932-0f3f-486c-acc6-e7d78b358f9b  Microsoft
[121]   TScienceGPU                           79f57c16-00fe-48da-87d4-5192e86cd047  Microsoft
[122]   UI Fabric                             bac044cf-49e1-4843-8dda-1ce9662606c8  Microsoft
[123]   WCD_AutomatedIR_Staging_LBI           41e8162b-dde8-493d-8b0a-2ec52f9f3d91  Microsoft
[124]   WCD_CloudServices_BE_Staging_LBI      8700d3a3-3bb7-4fbe-a090-488a1ad04161  Microsoft
[125]   WCD_Customer_DEV_Staging_LBI          ecaaecb5-5d9d-4eec-966e-003aa60c938b  Microsoft
[126]   WCD_Deception_Staging_LBI             21b96909-5e4d-433f-a883-2a1d7cc0d655  Microsoft
[127]   WCD_Disrupt_Staging_LBI               e59ec34c-ffad-4154-912c-14f35836d09e  Microsoft
[128]   WCD_INE_Staging_DPFT_LBI              73efafab-6b23-4a89-ab3e-fe6e36370f1a  Microsoft
[129]   WCD_INE_Staging_DSP_LBI               78fbf82d-36ae-4d2e-9018-cecd96c88fd1  Microsoft
[130]   WCD_INE_Staging_Hunt_LBI              59cad1fb-9b91-44a9-a396-adc9743ae620  Microsoft
[131]   WCD_INE_Staging_INR_LBI               b544616c-a4f0-423b-8452-d211e69f8a6b  Microsoft
[132]   WCD_InE_Staging_LBI                   f0474004-b2ce-45a3-aae3-5f18fa302401  Microsoft
[133]   WCD_INE_Staging_USE_LBI               3aadaa4f-082f-4e81-ba9e-4c3bdcb21763  Microsoft
[134]   WCD_Kusto_Supportabilty_Staging_LBI   4be6bda1-5a26-4212-9b3c-a5df137b0a67  Microsoft
[135]   WCD_Kusto_TVM_Staging_LBI             f3088921-457d-42d9-8157-fc4ce959892f  Microsoft
[136]   WCD_MicroServices_Staging_LBI         c5d1c552-a815-4fc8-b12d-ab444e3225b1  Microsoft
[137]   WCD_Monitoring_Staging_LBI            294592bd-2dcb-45aa-aafd-16ff2f5c6cf3  Microsoft
[138]   WCD_OPS_POCs_Staging_LBI              50822bc7-f43c-4d62-a868-1aaba3444e23  Microsoft
[139]   WCD_Sense_Staging_EUS_LBI             16a6c94a-cac2-4e68-bd0b-64daa28f428f  Microsoft
[140]   WCD_TIMNA_Staging_LBI                 99d06da1-455b-4a23-b2df-3181e98fe7bf  Microsoft
[141]   WCD_UEBA_Microservices_Staging_LBI    2d2760ec-3098-4911-b808-fb2b59df41b4  Microsoft
[142]   xt-teams-migration-dev                6760347d-9ffe-41a9-ba11-c139dcea5ce6  Microsoft

The default is marked with an *; the default tenant is 'Microsoft' and subscription is 'Azure Security Data Science Development' (0bd0a0b9-12e9-4949-9011-38eb480fdf64).

Select a subscription and tenant (Type a number or Enter for no changes):

Tenant: Microsoft
Subscription: Azure Security Data Science Development (0bd0a0b9-12e9-4949-9011-38eb480fdf64)

[Announcements]
With the new Azure CLI login experience, you can select the subscription you want to use more easily. Learn more about it and its configuration at https://go.microsoft.com/fwlink/?linkid=2271236

If you encounter any problem, please open an issue at https://aka.ms/azclibug

[Warning] The login output has been updated. Please be aware that it no longer displays the full list of available subscriptions by default.


(chatosp) C:\Users\t-grbabalola\PhishResearch\sources\DataScience\ChatOSP>python main.py
Running on local URL:  http://0.0.0.0:7860

To create a public link, set `share=True` in `launch()`.
Traceback (most recent call last):
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\queueing.py", line 541, in process_events
    response = await route_utils.call_process_api(
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\route_utils.py", line 276, in call_process_api
    output = await app.get_blocks().process_api(
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\blocks.py", line 1928, in process_api
    result = await self.call_function(
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\blocks.py", line 1526, in call_function
    prediction = await utils.async_iteration(iterator)
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\utils.py", line 657, in async_iteration
    return await iterator.__anext__()
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\utils.py", line 783, in asyncgen_wrapper
    response = await iterator.__anext__()
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\chat_interface.py", line 592, in _stream_fn
    first_response = await async_iteration(generator)
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\utils.py", line 657, in async_iteration
    return await iterator.__anext__()
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\utils.py", line 650, in __anext__
    return await anyio.to_thread.run_sync(
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\anyio\_backends\_asyncio.py", line 2177, in run_sync_in_worker_thread
    return await future
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\anyio\_backends\_asyncio.py", line 859, in run
    result = context.run(func, *args)
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\gradio\utils.py", line 633, in run_sync_iterator_async
    return next(iterator)
  File "C:\Users\t-grbabalola\PhishResearch\sources\DataScience\ChatOSP\gpt_agent.py", line 57, in my_response_wiki
    search_results = search_session.query_search_service(user_input)
  File "C:\Users\t-grbabalola\PhishResearch\sources\DataScience\ChatOSP\gpt_agent.py", line 20, in query_search_service
    for result in results:
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\azure\search\documents\_paging.py", line 54, in __next__
    return next(self._page_iterator)
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\azure\core\paging.py", line 75, in __next__
    self._response = self._get_next(self.continuation_token)
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\azure\search\documents\_paging.py", line 124, in _get_next_cb
    return self._client.documents.search_post(search_request=self._initial_query.request, **self._kwargs)
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\azure\core\tracing\decorator.py", line 94, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "C:\Users\t-grbabalola\AppData\Local\anaconda3\envs\chatosp\lib\site-packages\azure\search\documents\_generated\operations\_documents_operations.py", line 756, in search_post
    raise HttpResponseError(response=response, model=error)
azure.core.exceptions.HttpResponseError: () Authorization failed.
Code:
Message: Authorization failed.

**Comment by jiasli** [comment]
Duplicate of #29334
